apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/Priority.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright 2020 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Google.Cloud.Spanner.Data
+{
+    /// <summary>
+    /// The relative priority for requests.
+    /// </summary>
+    public enum Priority
+    {
+        /// <summary>
+        /// Executes the request with unspecified priority.
+        /// </summary>
+        Unspecified,
+        /// <summary>
+        /// Executes the request with low priority.
+        /// </summary>
+        Low,
+        /// <summary>
+        /// Executes the request with medium priority.
+        /// </summary>
+        Medium,
+        /// <summary>
+        /// Executes the request with high priority.
+        /// </summary>
+        High
+    }
+
+    internal class PriorityConverter
```

Comment:
This makes me nervous. We need to consider what will happen if a new value is added in the proto vs our enum here.
I would be tempted to keep things in terms of the Spanner.Data enum everywhere we can, so we only ever need to convert *to* the proto form. That would mean keeping it separately from the RequestOptions, admittedly.

Alternatively, we could potentially just convert numerically:

```csharp
internal static Priority FromProto(V1.RequestOptions.Types.Priority proto) => (Priority) (int) proto;
```

We'd need to keep the numeric values in sync, but I think that's fine. At that point, we really get transparency.

Reply:
+1 to keeping things separately from the RequestOptions. For this PR and the Tags one. And only construct the RequestOptions when they are going to be used.

Reply:
I've removed `RequestOptions` from the commands and transactions and introduced `Priority` as a separate property. Once we have agreement on this in this PR, I'll look into changing #5575 so they are in sync with each other.

I also changed the conversion of `Priority` to a pure numeric conversion.